### DISCUSSION TITLE
Revisiting the `Ready` status

### DISCUSSION THREAD

**Discussion by chanwit** [flux-iac/tofu-controller#320] [Ideas]
After adding features like Outputs and HealthChecks, the "Ready" status of the object has been changed unknowingly,  breaking the dependency in a Kustomization chain. 

Integration of TF-controller into the GitOps workflow must be first-class supported. It is important to get this behavior fixed and fixed correctly.

We'll discuss here what would be a good behaviour, including the state machine to make the behavior "right."

**Comment by nalum** [comment]
What did these change about it that it is now broken?

**Comment by chanwit** [comment]
@Nalum here it is https://github.com/weaveworks/tf-controller/issues/302

**Comment by chanwit** [comment]
After applying,  the controller should not make the `Ready` status `true` if Health Checks exist, for example.

**Reply by nalum** [reply]
Ah so there needs to be a state machine, as you noted, that tracks all the conditions and makes sure there is no condition that would contradict a `Ready` `true` condition, if there is then `Ready` should be `false`?

**Reply by chanwit** [reply]
Maybe `unknown` until it would be either `true` or `false`.

**Comment by chanwit** [comment]
Delivered in v0.13.0
